src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -1138,12 +843,14 @@ private SSTableReader cloneAndReplace(DecoratedKey newFirst, OpenReason reason,
                                                  metadata,
                                                  ifile != null ? ifile.sharedCopy() : null,
                                                  dfile.sharedCopy(),
-                                                 newSummary,
                                                  bf.sharedCopy(),
                                                  maxDataAge,
                                                  sstableMetadata,
                                                  reason,
-                                                 header);
+                                                 header,
+                                                 newSummary);
+
+        // TODO: encapsulate this in the builder
```

Comment:
@jasonstack `first` is a bit more complicated than `indexSummary` and friends to make final because of how the writer uses it during append. I suppose we can leave this as it is and remove the TODO.

Reply:
+1